herddb-core/src/main/java/herddb/model/ColumnTypes.java

```diff
@@ -81,4 +81,17 @@ public static boolean isNotNullDataType(int type) {
         }
     }
 
+    public static int getNonNullTypeForPrimitiveType(int type) {
+        switch(type) {
+            case STRING:
+                return NOTNULL_STRING;
+            case INTEGER:
+                return NOTNULL_INTEGER;
+            case LONG:
+                return NOTNULL_LONG;
+            default:
+                return type;
```

Comment:
not sure this is the best return value, what about am IllegalArgumentException ?
Other wide it is better to add a Javadoc

Reply:
I modified the method to add more types that are supported in ColumnTypes.